### ISSUE TITLE
Not working on Steam Deck

### ISSUE THREAD

**Issue by TotalMisplay** [rimpy-custom/RimPy#122]
I haven't used my Steam Deck in around about a month and I picked it back up to get it ready since I'm gonna be on the go for a while, but when I tried to load RimPy up I got this error instead:
ERROR: ld.so: object 'libpcre.so.3' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
Traceback (most recent call last):
 File "/home/deck/Programs/RimPy Mod Manager/RimPy_Linux/rimpy.py", line 26, in <module>
ImportError: /home/deck/Programs/RimPy Mod Manager/RimPy_Linux/PyQt5/../libnssutil3.so: version `NSSUTIL_3.82' not found (required by /usr/lib/libnss3.so)
I used RimPy on my Steam Deck before with no issue so I don't know why this is or how to fix it. Reinstalling did not help unfortunately and I don't know enough about Linux to figure this out on my own.

**Comment by matej-vrba** [comment]
I dont have steam deck, but 
- first of all make sure all software is up to date using GUI software store or `sudo pacman -Suy` in terminal
- if that doesn't help try installing `libpcre` using `sudo pacman -S pcre lib32-pcre`
- install nss - `sudo pacman -S nss python-nss lib32-nss`

You probably don't have to use the terminal. It should be possible to install the packages from GUI software store

**Comment by TotalMisplay** [comment]
> I dont have steam deck, but
> 
> * first of all make sure all software is up to date using GUI software store or `sudo pacman -Suy` in terminal
> * if that doesn't help try installing `libpcre` using `sudo pacman -S pcre lib32-pcre`
> * install nss - `sudo pacman -S nss python-nss lib32-nss`
> 
> You probably don't have to use the terminal. It should be possible to install the packages from GUI software store

Unfortunately the terminal is essentially worthless because the Steam Deck has its OS in Read Only mode, so every command like that returns an error. I tried regardless, but they did not work unfortunately. As for installing the packages from the GUI Software Store, could you elaborate a bit more? Treat me like I am 5 and drooling currently. What I do know is SteamOS is built off of ArchLinux, so I take it you mean the Discovery app?

**Comment by matej-vrba** [comment]
Ok, after taking a deeper look at Steam OS you have several options:
- Steam OS (as you mentioned) is in Read Only mode. You can disable this and update all software necessary, but accepting any OS update via Steam will likely undo all changes
- Use older version of RimPy
- Ask the maintainer of RimPy to make flatpak/appimage - see #118
- Ask the maintainer to make RimPy open source - this will make it easier for other people to make appimage/flatpak - see #79 (but make sure to put hazmat suit on first as the thread is quite toxic :))
- use a workaround from #87

If you decide for 1st, 2nd or 5th option I'll be happy to help you.
IMO 2nd option is best for you as it will be the easiest.

**Comment by TotalMisplay** [comment]
> Ok, after taking a deeper look at Steam OS you have several options:
> 
> * Steam OS (as you mentioned) is in Read Only mode. You can disable this and update all software necessary, but accepting any OS update via Steam will likely undo all changes
> * Use older version of RimPy
> * Ask the maintainer of RimPy to make flatpak/appimage - see [Add the support of flatpak and publish on flathub #118](https://github.com/rimpy-custom/RimPy/issues/118)
> * Ask the maintainer to make RimPy open source - this will make it easier for other people to make appimage/flatpak - see [Question: Why is RimPy closed source? #79](https://github.com/rimpy-custom/RimPy/issues/79) (but make sure to put hazmat suit on first as the thread is quite toxic :))
> * use a workaround from [Doesn't work with nss 3.82 #87](https://github.com/rimpy-custom/RimPy/issues/87)
> 
> If you decide for 1st, 2nd or 5th option I'll be happy to help you. IMO 2nd option is best for you as it will be the easiest.

Okie doke. If I wanted to use an older version, how older would you recommend? I am still really confused by this issue because RimPy WAS working on my Steam Deck a little over a month ago. I didn't even see that other issue with nss 3.82 despite me googling that exactly and with RimPy in the search. Thank you, Google, very cool.

**Comment by matej-vrba** [comment]
> If I wanted to use an older version, how older would you recommend?

idk looking at the change log [1.2.6.27](https://github.com/rimpy-custom/RimPy/releases/tag/1.2.6.27)

**Comment by matej-vrba** [comment]
> I didn't even see that other issue with nss 3.82 despite me googling that exactly and with RimPy in the search.

yeah, imo it's better to search in the [issue list](https://github.com/rimpy-custom/RimPy/issues), but you have to know what you're looking for

**Comment by TotalMisplay** [comment]
> > I didn't even see that other issue with nss 3.82 despite me googling that exactly and with RimPy in the search.
> 
> yeah, imo it's better to search in the [issue list](https://github.com/rimpy-custom/RimPy/issues), but you have to know what you're looking for

I noticed in that other issue you linked someone said "I downloaded nss-3.82 from here: https://archive.archlinux.org/packages/n/nss/, extracted to RimPy/nss, then changed the bash script to:" and provided what they changed their bash to. I was looking how to do it manually like this. Were this Windows I'd know what to do, but in the world of Linux I am a ity bity babby :pensive:

**Comment by matej-vrba** [comment]
Well. you can do that, if you want.
1. download [nss](https://archive.archlinux.org/packages/n/nss/)
2. in RimPy directory create nss directory
3. extract content of the archive from 1. to the nss directory
4. change 2nd line in `RimPy.sh` to `LD_PRELOAD="nss/libnssutil3.so:libpcre.so.3:$LD_PRELOAD" ./RimPy`

**Comment by TotalMisplay** [comment]
> Well. you can do that, if you want.
> 
> 1. download [nss](https://archive.archlinux.org/packages/n/nss/)
> 2. in RimPy directory create nss directory
> 3. extract content of the archive from 1. to the nss directory
> 4. change 2nd line in `RimPy.sh` to `LD_PRELOAD="nss/libnssutil3.so:libpcre.so.3:$LD_PRELOAD" ./RimPy`

It is still not working with the same error. Am I just brain damaged?

**Comment by matej-vrba** [comment]
- did you download version 3.82? 
- run `find . | nc termbin.com 9999` inside RimPy directory and send me the link it will return - this will show me how the directory looks (if you want to know what it will upload, just run `find .`

**Comment by TotalMisplay** [comment]
> * did you download version 3.82?
> * run `find . | nc termbin.com 9999` inside RimPy directory and send me the link it will return - this will show me how the directory looks (if you want to know what it will upload, just run `find .`

I downloaded "[nss-3.82-1-x86_64.pkg.tar.zst](https://archive.archlinux.org/packages/n/nss/nss-3.82-1-x86_64.pkg.tar.zst)"
I copied and pasted "find . | nc termbin.com 9999" after opening the terminal in the RimPy directory, but it said "nc: command not found" and I tried just using the search function in the folder itself but in my experience that just always turns up nothing no matter what. I have literally no idea what I am doing. I have never touched Linux before SteamOS.

**Comment by Mearsu** [comment]
Ok I have to go for now (so don't wait for my answer).
When I get back I'll delete this comment and hopefully finaly fix your issue.

Github should send you an email every time somebody replies here, so make sure it's not going to spam ;)

p.s. do not respond or quote this message, so this thread makes sense after I delete this :)

**Comment by TotalMisplay** [comment]
So, I tried a fresh download of RimPy aaaand...
ERROR: ld.so: object 'libpcre.so.3' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
Traceback (most recent call last):
  File "/home/deck/Programs/RimPy Mod Manager/rimpy.py", line 24, in <module>
ImportError: /home/deck/Programs/RimPy Mod Manager/PyQt5/QtCore.so: cannot open shared object file: No such file or directory

This is headache inducing...

**Comment by TotalMisplay** [comment]
I have absolutely no idea what happened. I extracted a fresh install 3 times. 3 times. On the third time I got a bunch of errors, but then RimPy opened. I don't claim to understand, I don't wish to understand. Thank you very much for helping me, @Mearsu and I am sorry I wasted your time. I have no idea what, how, or why, but I am going to smash my head against my wall now. I hope you have a very good day.
P.S. Multiple fresh installs, not sure why THIS time it worked.
